htdocs/societe/contact.php

```diff
@@ -138,23 +138,19 @@
 if ($action == 'confirm_delete' && $user->hasRight('societe', 'contact', 'delete')) {
 	$id = GETPOST('id', 'int');
 	if (!empty($id) && $socid > 0) {
-		$db->begin();
-
-		$sql = "DELETE t, et FROM ".MAIN_DB_PREFIX."socpeople AS t";
-		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."socpeople_extrafields AS et ON t.rowid = et.fk_object";
-		$sql .= " WHERE t.fk_soc = ".((int) $socid);
-		$sql .= " AND t.rowid = ".((int) $id);
-		$sql .= " AND ((t.fk_user_creat = ".((int) $user->id)." AND t.priv = 1) OR t.priv = 0)";
-
-		$result = $db->query($sql);
-		if (!$result) {
-			setEventMessages($db->lasterror(), null, 'errors');
-			$db->rollback();
-		} else {
-			$db->commit();
+		$contact = new Contact($db);
+
+		$result = $contact->fetch($id);
```

Comment:
There's no more error checking on the fetch() operation. Should it be checked also?